pkg/rpc/heartbeat.go

```diff
@@ -55,6 +55,12 @@ type HeartbeatService struct {
 
 	onHandlePing func(context.Context, *PingRequest) error // see ContextOptions.OnIncomingPing
 
+	// This is a blocking call that verifies the underlying TCP connection is
+	// established. If there is already a healthy connection set up, it will
+	// simply return immediately, however if not, it attempts to establish a
+	// "dummy" connection which is never used to send messages on.
+	verifyDialback func(context.Context, roachpb.NodeID) error
```

Comment:
We already have `onHandlePing` which effectively does the same thing, should we hook into that instead? We can't go via `ContextOptions.OnIncomingPing` because that's necessarily constructed before the RPC context itself, but we can inject a handler during RPC context construction before setting `onHandlePing`. Not particularly important, and I'm not sure if it's really making things more clear or less, so take it or leave it.